src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1127,30 +1148,28 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
       // we need to update the balance and nonce of the sender _before_
       // we run this transaction so that things that rely on these values
       // are correct (like contract creation!).
-      const fromAccount = await vm.stateManager.getAccount({
+      const fromAccount = await vm.eei.getAccount({
         buf: caller
       } as any);
-      fromAccount.nonce.iaddn(1);
-      const txCost = new BN(
-        (gasLimit * transaction.gasPrice.toBigInt()).toString()
-      );
-      fromAccount.balance.isub(txCost);
-      await vm.stateManager.putAccount({ buf: caller } as any, fromAccount);
+      fromAccount.nonce += 1n;
+      const txCost = gasLimit * transaction.gasPrice.toBigInt();
+      const startBalance = fromAccount.balance;
+      // TODO: should we throw if insufficient funds?
```

Comment:
In previous ejs versions, if we subtracted from a balance such that the result would be negative, they just parsed it as a positive number. So we never had to worry about a negative balance. Now, however, if we pass on a negative number ejs will throw. 

If we run the transaction with a zero balance, the vm will throw with insufficient funds anyways, so I think we can possibly remove this todo.